python/lsst/pipe/tasks/htmIndexEngine.py

```diff
@@ -0,0 +1,352 @@
+            record.setId(row[self.config.id_name])
+        else:
+            rec_num += 1
+            record.setId(rec_num)
+        # No parents
+        record.setParent(-1)
+
+        self._set_flags(record, row)
+        self._set_mags(record, row)
+        self._set_extra(record, row)
+        return rec_num
+
+    def get_catalog(self, dataId, schema):
+        """!Get a catalog from the butler or create it if it doesn't exist
+
+        @param[in] dataId  Identifier for catalog to retrieve
+        @param[in] schema  Schema to use in catalog creation if the butler can't get it
+        @param[out] afwTable.SourceCatalog for the specified identifier
+        """
+        if self.butler.datasetExists(self.config.ref_dataset_name, dataId=dataId):
+            return self.butler.get(self.config.ref_dataset_name, dataId=dataId)
+        return afwTable.SourceCatalog(schema)
+
+    def make_schema(self, dtype):
+        """!Make the schema to use in constructing the persisted catalogs.
+
+        @param[in] dtype  A numpy.dtype to use in constructing the schema
+        @param[out] The schema for the output source catalog.
+        """
+        mag_column_list = self.config.mag_column_list
+        mag_err_column_map = self.config.mag_err_column_map
+        if len(mag_err_column_map) > 0 and (not len(mag_column_list) == len(mag_err_column_map)
+                or not sorted(mag_column_list) == sorted(mag_err_column_map.keys())):
+            raise ValueError("Every magnitude column must have a corresponding error column")
+        # makes a schema with a coord, id and parent_id
+        schema = afwTable.SourceTable.makeMinimalSchema()
+
+        def add_field(name):
+            if dtype[name].kind == 'S':
+                # dealing with a string like thing.  Need to get type and size.
+                at_type = afwTable.aliases[str]
+                at_size = dtype[name].itemsize
+                return schema.addField(name, type=at_type, size=at_size)
+            else:
+                at_type = afwTable.aliases[dtype[name].type]
+                return schema.addField(name, at_type)
+
+        for item in mag_column_list:
+            key = schema.addField(item+'_flux', float)
+        if len(mag_err_column_map) > 0:
+            for err_item in mag_err_column_map.keys():
+                key = schema.addField(err_item+'_fluxSigma', float)
+        for flag in self._flags:
+            attr_name = 'is_{}_name'.format(flag)
+            if getattr(self.config, attr_name):
+                key = schema.addField(flag, 'Flag')
+        for col in self.config.extra_col_names:
+            key = add_field(col)
+        return schema
+
+
+class HtmIndexer(object):
+    def __init__(self, depth=8):
+        """!Construct the indexer object
+
+        @param[in] depth  depth of the hierarchy to construct
+        """
+        self.htm = esutil.htm.HTM(depth)
+        # HACK need to call intersect first otherwise it segfaults
```

Comment:
Yes.  I talked with Erin about it, but it is unresolved at the moment.  There is an [issue](https://github.com/esheldon/esutil/issues/12).